service/src/main/java/org/cbioportal/genome_nexus/service/cached/CachedVariantAnnotationFetcher.java

```diff
@@ -1,25 +1,97 @@
 package org.cbioportal.genome_nexus.service.cached;
 
+import com.mongodb.DBObject;
 import org.cbioportal.genome_nexus.model.VariantAnnotation;
 import org.cbioportal.genome_nexus.persistence.VariantAnnotationRepository;
 import org.cbioportal.genome_nexus.persistence.internal.VariantAnnotationRepositoryImpl;
+import org.cbioportal.genome_nexus.service.exception.ResourceMappingException;
 import org.cbioportal.genome_nexus.service.transformer.ExternalResourceTransformer;
 import org.cbioportal.genome_nexus.service.remote.VEPDataFetcher;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 @Component
 public class CachedVariantAnnotationFetcher extends BaseCachedExternalResourceFetcher<VariantAnnotation, VariantAnnotationRepository>
 {
+    public static final String FIXED_ENTRY = "AGT:c.803T>C";
+
     @Autowired
     public CachedVariantAnnotationFetcher(ExternalResourceTransformer<VariantAnnotation> transformer,
                                           VariantAnnotationRepository repository,
-                                          VEPDataFetcher fetcher)
+                                          VEPDataFetcher fetcher,
+                                          @Value("${vep.max_page_size}") Integer maxPageSize)
```

Comment:
@onursumer good to specify default here as well `${vep.max_page_size:300}`. This way it doesn't crash when u don't update your properties file